src/NitroSharp/Content/ArchiveManager.cs

```diff
@@ -0,0 +1,255 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Text;
+
+namespace NitroSharp.Content
+{
+    internal interface IArchiveFile : IDisposable
+    {
+        public void OpenArchive();
+        public Stream OpenStream(string path);
+        public bool Contains(string path);
+    }
+
+    internal class VFSNode : IDisposable
+    {
+        public List<IArchiveFile>? MountedArchives { get; set; }
+        public Dictionary<string, VFSNode>? Children { get; set; }
+
+        private bool _disposed = false;
+
+        ~VFSNode()
+        {
+            Dispose(false);
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    if (Children != null)
+                    {
+                        foreach (KeyValuePair<string, VFSNode> pair in Children)
+                        {
+                            pair.Value.Dispose();
+                        }
+                    }
+
+                    if (MountedArchives != null)
+                    {
+                        foreach (IArchiveFile mountedArchive in MountedArchives)
+                        {
+                            mountedArchive.Dispose();
+                        }
+                    }
+                }
+                _disposed = true;
+            }
+        }
```

Comment:
There's no need for a `Dispose` method of any kind — a node doesn't _own_ the archives it points to.

Reply:
That's not how I see it, no one else `Dispose` those since `ArchiveManager` only `Dispose` the root node. I'm probably missing something with my poor C# knowledge.

Reply:
No, you're not missing anything. I was wrong 🙂